### PR TITLE
[new release] sihl-web, sihl-email, sihl-persistence, sihl-type, sihl-contract, sihl-core, sihl-queue, sihl-user, sihl-storage and sihl (0.2.0)

### PR DESCRIPTION
**Pull request by jerben**
Contains HTTP server implementations as Sihl service

- Project page: <a href="https://github.com/oxidizing/sihl">https://github.com/oxidizing/sihl</a>
- Documentation: <a href="https://oxidizing.github.io/sihl/">https://oxidizing.github.io/sihl/</a>

### Fixed
- Extract `sihl-core`, `sihl-type`, `sihl-contract`, `sihl-user`, `sihl-persistence` and `sihl-web` as separate opam packages
- Increase session key size to 20 bytes
- Sign session cookies with `SIHL_SECRET`
- Simplify session service API
- Implement generic flash storage on top of session storage and replace the specific message service
- Update to httpaf-based Opium 0.19.0

### Added
- File log reporter to store logs in `logs/error.log` and `logs/app.log`
- Add log source to log text

### CODE REVIEW

File: packages/sihl-type/sihl-type.0.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+depends: [
+  "dune" {>= "2.7"}
+  "ocaml" {>= "4.08.0"}
+  "sihl-core" {= version}
+  "opium" {>= "0.19.0"}
+  "multipart-form-data" {>= "0.3.0"}
+  "uuidm" {>= "0.9.7"}
+  "alcotest" {>= "1.2.0"}
+  "alcotest-lwt" {>= "1.2.0" & < "3.0.0" & with-test}
+  "odoc" {with-doc}
```
Comment (mseri) [inline packages/sihl-type/sihl-type.0.2.0/opam:20]:
```suggestion
  "odoc" {with-doc}
  "caqti-lwt"
```

This is failing with
```
#=== ERROR while compiling sihl-type.0.2.0 ====================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.11.1 | pinned(https://github.com/oxidizing/sihl/releases/download/0.2.0/sihl-0.2.0.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/sihl-type.0.2.0
# command              ~/.opam/4.11/bin/dune build -p sihl-type -j 47 @install
# exit-code            1
# env-file             ~/.opam/log/sihl-type-1-c99ec6.env
# output-file          ~/.opam/log/sihl-type-1-c99ec6.out
### output ###
# File "sihl-type/src/dune", line 4, characters 49-58:
# 4 |  (libraries sihl-core ptime ptime.clock.os caqti caqti-lwt uuidm opium
#                                                      ^^^^^^^^^
# Error: Library "caqti-lwt" not found.
# Hint: try:
#   dune external-lib-deps --missing -p sihl-type -j 47 @install
```

Comment (jerben) [reply packages/sihl-type/sihl-type.0.2.0/opam:21]: Fixed, thank you!

**Comment by mseri** [comment]
There is something weird going on, could you have a look: https://opam.ci.ocaml.org/github/ocaml/opam-repository/commit/7acfccfa83df6f01c03c498237696d51de1bd896 ?

**Comment by jerben** [comment]
I pushed the wrong `caqti-lwt` version, should be fixed now.

**Comment by mseri** [comment]
Looks good now, thanks

**Pull request merged**